cpp/src/arrow/type.h

```diff
@@ -123,6 +123,9 @@ struct Type {
     /// A list of some logical data type
     LIST,
 
+    /// A list of some logical data type, using 64-bit offsets
+    LARGE_LIST,
```

Comment:
Putting this before the end of the `enum` will break the ABI. I'm not sure we care. @xhochy Comments ?

Reply:
We sadly break the ABI quite a lot, I've given up a bit on keeping it stable. I'll look into this in 3-6 months again.